src/angular/searchbox/search-bar.component.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Created by rc2122 on 11/15/2017.
+ */
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+    selector: 'sdc-search-bar',
+    templateUrl: './search-bar.component.html'
+})
+export class SearchBarComponent {
+
+    @Input() public placeholder: string;
+    @Input() public label: string;
+    @Input() public searchQuery: string;
+    @Input() public debounceTime: number = 200;
+    @Output() public searchChanged: EventEmitter<any> = new EventEmitter<any>();
+    @Output() public searchButtonClicked: EventEmitter<string> = new EventEmitter<string>();
+
```

Comment:
@lizronen @michalweinsh @einavk 
Liz, Michal and Einav please tell what do you think

It is not clear why we have: searchChanged and searchButtonClicked.

It looks like these are 2 different components:

1. **Search bar**: 
- perform action on click. 
- magnify change to "X" after clicking magnify (and not while typing).
- has event searchButtonClicked

2. **Filter bar** 
- perform action while writing. 
- magnify change to "X" after typing value.
- has event searchChanged (2 ways binding)